CMakeLists.txt

```diff
@@ -153,35 +171,48 @@ include(CPack)
 message(STATUS "Setting link libraries")
 
 function(protocol protoPath protoName external)
-    if (external)
+    if(external)
         execute_process(
             COMMAND ${WaylandScanner} server-header ${protoPath} protocols/${protoName}-protocol.h
             WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         execute_process(
             COMMAND ${WaylandScanner} private-code ${protoPath} protocols/${protoName}-protocol.c
-            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})  
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         target_sources(Hyprland PRIVATE protocols/${protoName}-protocol.c)
     else()
         execute_process(
             COMMAND ${WaylandScanner} server-header ${WAYLAND_PROTOCOLS_DIR}/${protoPath} protocols/${protoName}-protocol.h
             WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         execute_process(
             COMMAND ${WaylandScanner} private-code ${WAYLAND_PROTOCOLS_DIR}/${protoPath} protocols/${protoName}-protocol.c
-            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})  
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
         target_sources(Hyprland PRIVATE protocols/${protoName}-protocol.c)
     endif()
 endfunction()
 
-target_link_libraries(Hyprland PkgConfig::deps)
-
 target_link_libraries(Hyprland
-        ${CMAKE_SOURCE_DIR}/subprojects/wlroots/build/libwlroots.so.12032 # wlroots is provided by us
-        OpenGL::EGL
-        OpenGL::GL
-        Threads::Threads
-        ${CMAKE_SOURCE_DIR}/subprojects/udis86/build/libudis86/liblibudis86.a
+    rt
+    PkgConfig::deps
+    OpenGL::EGL
+    OpenGL::GL
+    Threads::Threads
+    wlroots
+    udis86
 )
 
+install(TARGETS Hyprland hyprctl DESTINATION bin)
+install(FILES ${wlroots_LIBRARY} DESTINATION lib)
+install(DIRECTORY assets/ DESTINATION share/hyprland PATTERN "meson.build" EXCLUDE)
+install(FILES example/hyprland.desktop DESTINATION share/wayland-sessions)
+install(FILES example/hyprland.conf DESTINATION share/hyprland)
+install(FILES docs/hyprctl.1 docs/Hyprland.1 DESTINATION share/man/man1)
+
+install(DIRECTORY ${wlroots_INCLUDE_DIR}/ DESTINATION include/hyprland/wlroots)
+install(DIRECTORY protocols/ DESTINATION include/hyprland/protocols PATTERN "meson.build" EXCLUDE)
+
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/hyprland.pc
```

Comment:
what is the prefix here? generally hyprland installs itself into `/usr/local`, but this has to be `/usr` (cuz arch)

Reply:
It's `/usr/local/` still, but that's fine, because directories under there have (almost) identical meanings to `/usr`, and packagers can override the prefix.

Reply:
pkg-config is the issue here, it doesn't include /usr/local for some reason on arch, so we want to install this one to /usr.

Reply:
That will be handled in the PKGBUILD, see: https://github.com/hyprwm/aur/blob/5e0fce6d9454d8ffe7e3e9ac50b37854ccec1b55/PKGBUILD-git#L84

Reply:
still don't believe this is a good idea, we should do both

Reply:
1. Then those packages need to be fixed.
2. Installing manually is a bad idea (and if someone wants to do it, they can just add the correct arguments to CMake)